### ISSUE TITLE
Will protection set in minutes override autolock until end of candle in live-run?

### ISSUE THREAD

**Issue by tonymorony** [freqtrade/freqtrade#6338]
## Describe your environment

  * Operating system: Ubuntu 18.04
  * Python Version: Python 3.8.10 
  * CCXT version: ccxt==1.70.45
  * Freqtrade Version: freqtrade develop-2e2b1e247

## Your question

Will protection set in minutes override autolock in live-run? 

e.g. I'm trading on 4h candles and trade is closing due to profitable trailing stop-loss, after that (if I understand correctly) I'll not able to buy in this candle even if the buy signal will happen
` Pair XXX/XXX is still locked until 2022-02-01 08:00:00 due to Auto lock.`

The question is will such protection override auto lock? :

```
        {
            "method": "StoplossGuard",
            "lookback_period_minutes": 5,
        }
 ```
 
the scenario is that trailing stop loss just fixed profits due to a reversal inside the candle but indicators are still bullish so I want a return to long as soon as receiving buy signal, and don't want to wait for another 4 hours :)
I'm backtesting with `--timeframe-detail 1m` detalization and such movements happens quite often it seems with trailing stop

**Comment by xmatthias** [comment]
Auto-lock until the end of the candle is non-optional.
Protections apply on top of that (if they are longer).

Auto-lock is there to protect against the scenario you're describing. If you think about it - what you're trying to do is a buy/sell cycle intra-candle.
You can get around this by using shorter candle timeframes - or by widening the gap until the stop.

The risk of having a buy/sell cycle when selling while the buy signal is still active is quite high.

Youll buy, stop out, buy again, stop out again, buy again, stop out again, ... is quite high - and can cost you real money if you're unlucky (i had a dry-run bot do that to me 20 times within a 15m candle on HOT/USDT, and it happened to others, too)

**Comment by tonymorony** [comment]
scenario visualisation:

1) Trailing stoploss sold in profit at 15:54

![Screenshot 2022-02-01 at 13 09 27](https://user-images.githubusercontent.com/24797699/151920917-4a5498fb-1a9f-49f1-8e4f-e0b2314de1a3.png)

2) Bought again at 16:00

![Screenshot 2022-02-01 at 13 09 54](https://user-images.githubusercontent.com/24797699/151920996-7ba68ca1-99b5-4122-a4d7-651b7db7dc82.png)
 
3) Sold on another trailing stoploss in profit at 16:08 

![Screenshot 2022-02-01 at 13 10 42](https://user-images.githubusercontent.com/24797699/151921075-4ca17681-e936-43fe-aae9-3b893950d86c.png)


`Auto-lock is there to protect against the scenario you're describing. If you think about it - what you're trying to do is a buy/sell cycle intra-candle.`

my actual stoploss is at -0.99 so trailing can trigger only in profit, not sure what are the risks here

```
   stoploss = -0.99

    # Trailing stop:
    trailing_stop = True
    trailing_stop_positive = 0.01
    trailing_stop_positive_offset = 0.079
    trailing_only_offset_is_reached = True
 ```
 
in backtests, with 1m detalisation it's lowering profits 3 times if I set such protector for auto-lock simulation (by default backtesting not considering auto-lock since making trades on single candle as shown above)

```
        {
            "method": "CooldownPeriod",
            "stop_duration_candles": 1
        }
 ```

**Comment by xmatthias** [comment]
in backtesting, you'll also not encounter multiple buys per candle even if the sell happens at the same candle - so while no explicit auto-lock is set - it's still there and the earliest next entry is in the next candle.
No longer auto-locking would make it work completely different to backtesting - which is a different risk altogether (but still something we'll try to avoid).

While it may be correct in your case - it's incorrect for most other cases - and therefore not something we'll be changing.

It's pretty easy to work around this by using a lower timeframe, as the auto-lock will always lock until the end of the candle - which is "timeframe" long - so to only lock for  a maximum of 15 minutes - use a timeframe of 15m.

You can still use informative pairs to base signals on the 4h candle (or whatever other candle you'd like).
